src/ff/field.rs

```diff
@@ -59,6 +59,7 @@ pub trait Field:
     /// Derived from the size of the backing field, this constant indicates how much
     /// space is required to store this field value
     const SIZE_IN_BYTES: u32 = Self::Integer::BITS / 8;
+    const STR_REPR: &'static str;
```

Comment:
this is confusing - it implies that there is a string representation of field **instance**, but it is actually a type id. Adding it to field does not prevent human mistakes because there is no centralized place to enforce uniqueness and the strings are sprinkled over different modules. 

I would suggest that we keep this mapping (string -> field type) in a function instead. Currently nobody uses that so network layer is probably the right place for it

Reply:
your point is valid, now I am not sure how I feel about leaving it out. Field is not an enum so we won't be able to get compile errors if someone adds a new type and forgets to update the mapping. Lmk if you want to return it or I can make that change too

Reply:
let me try adding a method to the `Field` trait that would do this mapping. It still won't be enforced by the compiler, but maybe it'll be fairly obvious that you _should_ add it if it's centralized